src/macaron/slsa_analyzer/checks/detect_malicious_metadata_check.py

```diff
@@ -22,31 +24,26 @@
 from macaron.slsa_analyzer.analyze_context import AnalyzeContext
 from macaron.slsa_analyzer.checks.base_check import BaseCheck
 from macaron.slsa_analyzer.checks.check_result import CheckResultData, CheckResultType, Confidence, JustificationType
-from macaron.slsa_analyzer.package_registry.pypi_registry import PyPIRegistry
+from macaron.slsa_analyzer.package_registry.pypi_registry import PyPIPackageJsonAsset, PyPIRegistry
 from macaron.slsa_analyzer.registry import registry
 
 logger: logging.Logger = logging.getLogger(__name__)
 
 
-class HeuristicAnalysisResultFacts(CheckFacts):
+class MaliciousMetadataFacts(CheckFacts):
     """The ORM mapping for justifications in pypi heuristic check."""
 
     __tablename__ = "_detect_malicious_metadata_check"
 
     #: The primary key.
     id: Mapped[int] = mapped_column(ForeignKey("_check_facts.id"), primary_key=True)  # noqa: A003
 
-    #: List of heuristic names that failed.
-    heuristics_fail: Mapped[str] = mapped_column(String, nullable=False, info={"justification": JustificationType.TEXT})
-
     #: Detailed information about the analysis.
-    detail_information: Mapped[str] = mapped_column(
-        String, nullable=False, info={"justification": JustificationType.TEXT}
-    )
+    detail_information: Mapped[dict[str, JsonType]] = mapped_column(DBJsonDict, nullable=False)
 
-    #: The result of heuristic analysis.
-    heuristic_result: Mapped[str] = mapped_column(
-        String, nullable=False, info={"justification": JustificationType.TEXT}
+    #: The result of analysis, which is of dict[Heuristics, HeuristicResult] type.
+    result: Mapped[dict[str, JsonType]] = mapped_column(
```

Comment:
Could you let me know more about the reason why we don't use `dict[Heuristics, HeuristicResult]` directly as the annotation, but we have to instead use `dict[str, JsonType]` ?

Reply:
Actually,  I noticed that the comment should change to `dict[str, JsonType]`. And the reason for not using `dict[Heuristics, HeuristicResult]` is because `HeuristicResult` is not a DB supported type.

Reply:
Yes, I can also use `dict[Heuristics, HeuristicResult]` since `HeuristicResult` has a `str` type that is supported by the DB.

Reply:
See commit [882e4f0](https://github.com/oracle/macaron/pull/797/commits/882e4f0c8d07db37001375d6c2d8bf5b9a64ae3d)